Kitodo/src/test/java/org/kitodo/production/helper/tasks/NewspaperMigrationTaskIT.java

```diff
@@ -0,0 +1,297 @@
+
+    private static final File METADATA_DIRECTORY = new File("src/test/resources/metadata");
+    private static final File ISSUE_ONE_PROCESS_DIR = new File(METADATA_DIRECTORY, "1");
+    private static final File ISSUE_TWO_PROCESS_DIR = new File(METADATA_DIRECTORY, "2");
+    private static final File ORIGINAL_METADATA_TEMPORARY_LOCATION = new File("metadata.orig");
+
+    private static final String MODS_MODS_XMLNS_MODS_HTTP_WWW_LOC_GOV_MODS_V3_MODS_EXTENSION = "<mods:mods xmlns:mods=\"http://www.loc.gov/mods/v3\"><mods:extension>";
+    private static final String GOOBI_GOOBI_XMLNS_GOOBI_HTTP_META_GOOBI_ORG_V1_5_1 = "<goobi:goobi xmlns:goobi=\"http://meta.goobi.org/v1.5.1/\">";
+    private static final String GOOBI_METADATA_NAME_TITLE_DOC_MAIN_NEWSPAPER_MIGRATION_TEST_GOOBI_METADATA = "<goobi:metadata name=\"TitleDocMain\">Newspaper migration test</goobi:metadata>";
+    private static final String METS_MPTR_LOCTYPE_URL_XLINK_HREF_XMLNS_XLINK_HTTP_WWW_W3_ORG_1999_XLINK = "<mets:mptr LOCTYPE=\"URL\" xlink:href=\"\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"/>";
+    private static final String GOOBI_GOOBI_MODS_EXTENSION_MODS_MODS_METS_XML_DATA_METS_MD_WRAP_METS_DMD_SEC = "</goobi:goobi></mods:extension></mods:mods></mets:xmlData></mets:mdWrap></mets:dmdSec>";
+
+    private static final BatchService batchService = ServiceManager.getBatchService();
+    private static final ProcessService processService = ServiceManager.getProcessService();
+
+    @BeforeClass
+    public static void prepareDatabase() throws Exception {
+        moveOriginMetadataDirectoryAside();
+        createMetadataDirectoryForNewspaperMigrationTest();
+        MockDatabase.startNode();
+        MockDatabase.insertProcessesFull();
+        addNewspaperDatabase();
+        User user = ServiceManager.getUserService().getById(1);
+        SecurityTestUtils.addUserDataToSecurityContext(user, 1);
+    }
+
+    private static void addNewspaperDatabase() throws Exception {
+        Process processOne = processService.getById(1);
+        Ruleset ruleset = processOne.getRuleset();
+        ruleset.setFile("ruleset_newspaper-migration-test.xml");
+        ServiceManager.getRulesetService().save(ruleset);
+        Process processTwo = processService.getById(2);
+        Batch newspaperBatch = new Batch();
+        newspaperBatch.setTitle("Newspaper migration test");
+        newspaperBatch.getProcesses().add(processOne);
+        newspaperBatch.getProcesses().add(processTwo);
+        ServiceManager.getBatchService().save(newspaperBatch);
+        processOne.setTitle("NewsMiTe_18500312");
+        processOne.getBatches().add(newspaperBatch);
+        processService.save(processOne);
+        processTwo.setTitle("NewsMiTe_18501105");
+        processTwo.getBatches().add(newspaperBatch);
+        processService.save(processTwo);
+    }
+
+    private static void moveOriginMetadataDirectoryAside() throws Exception {
+        if (ORIGINAL_METADATA_TEMPORARY_LOCATION.exists()) {
```

Comment:
why do you move the whole metadata directory?
Shouldn't you just restore the newspaper folders afterwards?

Reply:
Because the task creates additional processes (annual process, overall process). Nothing must be in the way.